### ISSUE TITLE
'Stream' input/output support for azure Blob functions seems broken

### ISSUE THREAD

**Issue by ewarwick** [Azure/azure-functions-dotnet-worker#2319]
### Description

Behavior: using a Stream as input crashes the function, using it as output writes 'System.IO.MemoryStream'  rather than the contents of the stream
Expected behavior:  Both should work as indicated in the documentation
OS: Ubuntu 22.04
SDK: Dotnet 8
Function app type: Isolated Process
Trigger: BlobTrigger
Example Repo: https://github.com/ewarwick/broken-blob-trigger

The documentation states that a Stream SHOULD be used as an output type for most files: [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-output?tabs=python-v2%2Cisolated-process%2Cnodejs-v4&pivots=programming-language-csharp)
It also states that Stream SHOULD be used as an input type for large files to avoid reading the whole thing into memory: [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-trigger?tabs=python-v2%2Cisolated-process%2Cnodejs-v4&pivots=programming-language-csharp)

> Binding to string, or Byte[] is only recommended when the blob size is small. This is recommended because the entire blob contents are loaded into memory. For most blobs, use a Stream or BlobClient type. 

However attempting to use the 'Stream' as an output results in a file with only the contents:
`System.IO.MemoryStream`

Attempting to use a 'Stream' as an input on the 'blobtrigger' results in the function failing to convert it and crashing before beginning to execute the body of the Function. It appears to be reading the entire contents into memory anyway and then attempting to parse it as JSON for some reason:
``` 
Exception: Microsoft.Azure.Functions.Worker.FunctionInputConverterException: Error converting 1 input parameters for Function 'DoTheThing': Cannot convert input parameter 'dropFile' to type 'System.IO.Stream' from type 'System.ReadOnlyMemory`1[[System.Byte, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'. Error:System.Text.Json.JsonException: 'D' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0.

```

### Steps to reproduce

- create an azure function with a BlobTrigger going into a variable of type Stream
- create an azure function with a return type of Stream and the BlobOutput attribute
- attempt to execute
- can also use the repro repo here: https://github.com/ewarwick/broken-blob-trigger

**Comment by ewarwick** [comment]
Update - This seems to be caused by expressions in the trigger and/or output. Seems like a pretty big thing to be broken if i understand correctly: this precludes having a single function which deals with large files of differing names - we would be forced to make a bunch of functions with hard-coded filenames in order to NOT read the entire file into memory?

https://github.com/Azure/azure-functions-dotnet-worker/issues/1969#issuecomment-1787637965

**Comment by domv3** [comment]
I was having a similar issue when I moved my blob trigger from .NET 6 to .NET 8 Isolated. Every time I would test my function it would come back with this cannon convert input parameter 'stream' to System.IO.Stream. The error right after that stated that it couldn't find my blob connection and that it doesn't exist. I had to add user secrets into my program.cs for it to work, like so.
After I added that, it never gave me that error again.
`.ConfigureAppConfiguration(config =>
{
    config.AddUserSecrets<YourFunctionName>(optional: true, reloadOnChange: false);
})`
